Common++/header/PointerVector.h

```diff
@@ -12,14 +13,23 @@
  */
 namespace pcpp
 {
+	/**
+ 	* @struct nullMutex
+ 	* A struct containing standard mutex operation but does not perform any action.
+ 	* This mutex is used when the user decides PointerVector not be thread safe which be default
+	*/
+	struct nullMutex {
+		void lock() const {}
+		void unlock() const {}
+	};
 
 	/**
 	 * @class PointerVector
 	 * A template class for representing a std::vector of pointers. Once (a pointer to) an element is added to this vector,
 	 * the element responsibility moves to the vector, meaning the PointerVector will free the object once it's removed from the vector
 	 * This class wraps std::vector and adds the capability of freeing objects once they're removed from it
 	 */
-	template<typename T>
+	template<typename T, typename Mutex=pcpp::nullMutex>
```

Comment:
please update the document.

Reply:
changed to NullMutex. But detail namespace is only present on Pcap++. Since this on Common folder hoping will place it here which is more reasonable.

Reply:
You can update the comment, mentioning about the template argument. We use Doxygen to generate the document.

Reply:
done